pennylane/math/multi_dispatch.py

```diff
@@ -248,15 +247,19 @@ def concatenate(values, axis=0):
         else:
             values = [torch.as_tensor(t) for t in values]
 
-    if interface == "tensorflow" and axis is None:
+    if like == "tensorflow" and axis is None:
         # flatten and then concatenate zero'th dimension
         # to reproduce numpy's behaviour
         values = [np.flatten(np.array(t)) for t in values]
         axis = 0
 
-    return np.concatenate(values, axis=axis, like=interface)
+    return np.concatenate(values, axis=axis, like=like)
 
 
+# Note that diag is not eligible for the multi_dispatch decorator because
+# it is used sometimes with iterable `values` that need to be interpreted
+# as a list of tensors, and sometimes with a single tensor `values` that
+# might not be iterable (for example a TensorFlow `Variable`)

```

Comment:
@dwierichs is it possible to modify the decorator to allow for this?

Reply:
I'm not sure. It would have to `try` to iterate over the arguments in (via the list method `extend`) that are marked via `tensor_list`, and if it fails, attempt to just `append` the respective argument to the arguments to dispatch over.
I'd consider this an unreasonable overhead given the currently single use case. What do you think?

Reply:
Oh, I was thinking something simpler, although maybe it is not in the spirit of 'multiple dispatch'.

Basically, check `not isinstance(arg, (list, tuple))`, and if this is the case, call `qml.math.get_interface(arg)`.

Basically, we assume that a `tensor_list` argument must be a built-in Python sequence (either list or tuple), and if this is not the case, we simply single dispatch

Reply:
Ah I see, that sounds cool! I'll give it a go :)

Reply:
I implemented this now, a bit more lightweight, even: If an argument marked as tensor list is not a `tuple` or `list`, it is simply treated as if it was not marked as a tensor list.

Reply:
ah, nice call!